### ISSUE TITLE
Optional arguments in blocks

### ISSUE THREAD

**Issue by lulco** [efabrica-team/phpstan-latte#335]
I think we shouldn't follow this behavior from latte. For static analysis is important to have correct types defined in methods, so if user writes this:
```latte
{define my-block, string $stringParam}
    // ...
{/define}
```
we should analyse $stringParam as required string parameter and not optional string|null parameter. It causes many false positives in a wild.

If user wants some parameter to be optional, it can be marked with default value. In actual state there is not such possibility. 

I'll try to prepare PR to change this behavior.

**Comment by MartinMystikJonas** [comment]
I wa thinking about the same when I encountered related false-positives on some of our projects. It might create different type of false positives but imho it is indeed better to respect explicitly set types.
